Engine/Authentication/AuthenticationSettings.cs

```diff
@@ -103,9 +116,35 @@ void PrepareRequest(HttpRequestMessage request, string domain, CancellationToken
         public HttpClient GetClient(string domain = null, bool withRetryPolicy = false)
         {
             var client = new HttpClient(new AuthenticationClientHandler(domain, withRetryPolicy));
-            if (Uri.IsWellFormedUriString(BaseAddress, UriKind.Absolute))
-                client.BaseAddress = new Uri(BaseAddress, UriKind.Absolute);
-            client.DefaultRequestHeaders.Add("User-Agent", $"OpenTAP/{PluginManager.GetOpenTapAssembly().SemanticVersion}");
+            if (BaseAddress != null)
+                client.BaseAddress = BaseAddress;
+            if(userAgent == null)
+            {
+                //var sw = System.Diagnostics.Stopwatch.StartNew();
+                userAgent = $"OpenTAP/{PluginManager.GetOpenTapAssembly().SemanticVersion}";
+                if(Cli.CliActionExecutor.SelectedAction is ITypeData td)
+                {
+                    // We are running a CLI Action. Add it's name and version to the User-Agent header
+                    var source = TypeData.GetTypeDataSource(td);
+                    userAgent += $" {td.Name}/{source.Version}";
+                }
+                else
+                {
+                    var asm = Assembly.GetEntryAssembly();
```

Comment:
As everything is `ICliActions` (REST-API, PackageRepo etc.) this will almost always be `tap.exe`.